### PR TITLE
feat: handle `JSON.parse` error

### LINKED ISSUE #222: Bug: "[object Object]" is not valid JSON
### Describe the bug

```
[E] novelai SyntaxError: "[object Object]" is not valid JSON
                            at JSON.parse (<anonymous>)
                            at request (D:\develop\koishi-dev\node_modules\koishi-plugin-novelai\lib\index.js:386:41)
                            at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
                            at async iterate (D:\develop\koishi-dev\node_modules\koishi-plugin-novelai\lib\index.js:424:31)
                            at async _Command.<anonymous> (D:\develop\koishi-dev\node_modules\koishi-plugin-novelai\lib\index.js:476:17)
                            at Array.<anonymous> (D:\develop\koishi-dev\node_modules\@koishijs\core\src\command\command.ts:299:14)
                            at _Command.execute (D:\develop\koishi-dev\node_modules\@koishijs\core\src\command\command.ts:315:22)
                            at <anonymous> (D:\develop\koishi-dev\node_modules\@koishijs\core\src\session.ts:397:22)
                            at Proxy.withScope (D:\develop\koishi-dev\node_modules\@koishijs\core\src\session.ts:293:22)
                            at next (D:\develop\koishi-dev\node_modules\@koishijs\core\src\middleware.ts:259:16)
```

### Steps to reproduce

`nai girl`

### Expected behavior

no error

### Screenshots

_No response_

### Relevant log output

_No response_

### Launcher

Koishi CLI (koishi start)

### Backend

Stable Diffusion WebUI (AUTOMATIC1111)

### Versions

- OS: Windows 11
- Adapter: Red
- Node version: 21.5.0
- Koishi version: 4.16.4

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by idanran**
fix #222

### CODE REVIEW

**Comment by MaikoTan** [comment]
If you are going to catch that error, you MUST handle it rather than leave it blank in the `catch` block.

**Comment by idanran** [comment]
> If you are going to catch that error, you MUST handle it rather than leave it blank in the `catch` block.

changed

**Review by MaikoTan — APPROVED** [review]

**Review by shigma — COMMENTED** [review]
你是不是应该处理它，而不是单纯 warn 一下？这里的 res.data.info 是不可解析的吗？

**Comment by idanran** [comment]
> 你是不是应该处理它，而不是单纯 warn 一下？这里的 res.data.info 是不可解析的吗？

更改了

**Comment by MaikoTan** [comment]
> `res.data.info` 有两种意外的类型，一个是 undefined，另一个是 object

Then what does `JSON.parse()` do, suppose that it can only be undefined or object?

**Comment by idanran** [comment]
> > `res.data.info` 有两种意外的类型，一个是 undefined，另一个是 object
> 
> Then what does `JSON.parse()` do, suppose that it can only be undefined or object?

throws an error

File: src/index.ts [hunk 1]
```diff
@@ -393,8 +393,18 @@ export function apply(ctx: Context, config: Config) {
...
-            return forceDataPrefix((res.data as StableDiffusionWebUI.Response).images[0])
+            const data = res.data as StableDiffusionWebUI.Response
+            if (data?.info?.prompt) {
+              finalPrompt = data.info.prompt
+            } else {
+              try {
+                finalPrompt = (JSON.parse(data.info)).prompt
+              } catch (err) {
+                const output = session.resolve(options.output ?? config.output)
+                if (output !== 'minimal') logger.warn(err)
```
Comment (shigma) [inline src/index.ts:404]:
```suggestion
                logger.warn(err)
```

`output` is not relavent to logger.

**Review by shigma — CHANGES_REQUESTED** [review]

**Pull request merged**